api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1317,50 +1373,43 @@ private Response execute(
         method, bucketName, objectName, region, headerMultiMap, queryParamMultiMap, body, length);
   }
 
-  /** Updates Region cache for given bucket. */
-  private void updateRegionCache(String bucketName)
+  /** Returns region of given bucket either from region cache or set in constructor. */
+  private String getRegion(String bucketName)
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    if (bucketName != null
-        && this.accessKey != null
-        && this.secretKey != null
-        && !BucketRegionCache.INSTANCE.exists(bucketName)) {
-      Map<String, String> queryParamMap = new HashMap<>();
-      queryParamMap.put("location", null);
-
-      Response response =
-          execute(Method.GET, bucketName, null, US_EAST_1, null, queryParamMap, null, 0);
-
-      String region;
-      try (ResponseBody body = response.body()) {
-        LocationConstraint lc = Xml.unmarshal(LocationConstraint.class, body.charStream());
-        if (lc.location() == null || lc.location().equals("")) {
-          region = US_EAST_1; // default region
-        } else if (lc.location().equals("EU")) {
-          region = "eu-west-1"; // eu-west-1 can be sometimes 'EU'.
-        } else {
-          region = lc.location();
-        }
-      }
+    if (this.region != null && !this.region.equals("")) {
+      return this.region;
+    }
 
-      // Add the new location.
-      BucketRegionCache.INSTANCE.set(bucketName, region);
+    if (!isAwsHost || bucketName == null || this.accessKey == null) {
+      return US_EAST_1;
     }
-  }
 
-  /** Returns region of given bucket either from region cache or set in constructor. */
-  private String getRegion(String bucketName)
-      throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
-          InternalException, InvalidBucketNameException, InvalidKeyException,
-          InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    String region;
-    if (this.region == null || "".equals(this.region)) {
-      updateRegionCache(bucketName);
-      region = BucketRegionCache.INSTANCE.region(bucketName);
-    } else {
-      region = this.region;
+    String region = AwsRegionCache.INSTANCE.get(bucketName);
+    if (region != null) {
+      return region;
+    }
+
+    // Updates region cache for this bucket.
```

Comment:
This comment is probably more appropriate just before the line

`AwsRegionCache.INSTANCE.set(bucketName, region);`

Reply:
I changed the comment to `// Execute GetBucketLocation REST API to get region of the bucket.` for specific meaning.